### ISSUE TITLE
Better display of Agent trajectory in front-end: allow go back in time with a slider

### ISSUE THREAD

**Issue by xingyaoww** [OpenHands/OpenHands#1743]
**What problem or use case are you trying to solve?**

**Describe the UX of the solution you'd like**

Related to https://github.com/OpenDevin/OpenDevin/issues/1742.
Similar to Devin, it supports visualizing a past state by dragging the button slider:

![image](https://github.com/OpenDevin/OpenDevin/assets/38853559/f9dc2ee1-a43a-4663-9068-2986513cad98)

**Do you have thoughts on the technical implementation?**

**Describe alternatives you've considered**

**Additional context**

**Comment by amanape** [comment]
Do you think we can utilize Git to help manage this state and diff (#1742)?

**Comment by rbren** [comment]
`docker commit` might be a better option here. That way you capture system-wide effects

I'm hesitant to use `git`, because it could interfere with the user's own git repo/commits/etc. It also would only capture code-level changes (not e.g. installing global software or a new dependency).

**Comment by Deng-Xian-Sheng** [comment]
I don't think this is very useful. For open source software, Devin's interface is too fancy.

I can't wait to reverse it now, and then take out the core part to see how they make LLM work together.

I guess, vectors, complex processes, and one LLM do only one thing.

They may abstract and subdivide the tasks, and then use different processes, or let LLM make the process.

The calling tool is just the end part, and decision-making may be an important part.

It should not be implemented by one LLM in one context, it can't be completed.

I wonder if there is a fast prototyping tool?

I want to focus on the cooperation between LLMs, not on writing code.

Then, design something fun. Even if there is no beautiful UI.

UI is not a moat.
